### PR TITLE
WL-3992 Added option to display root folder for the resources.

### PR DESCRIPTION
**Pull request by ouit0408**
Have added attribute 'displayRootDirectory' which default is set to false
but when set true will display the root folder.
Should not break Lawrence's code.

### CODE REVIEW

File: library/src/webapp/editor/ckextraplugins/common-wl/js/folder-listing.js [hunk 1]
```diff
@@ -28,9 +29,20 @@ $.fn.folderListing = function(options) {
 
   var urlPrefix = '/direct/content/resources/';
 
-  var getHtmlResults = function(json, $div) {
+  var getHtmlResults = function(json, $div, root) {
     var html = $('<ul/>').addClass('jqueryFileTree').hide();
-
+    var urlRoot = json[0].resourceId.trim();
+    if(settings.displayRootDirectory && urlRoot === root ){
```
Comment (buckett) [inline library/src/webapp/editor/ckextraplugins/common-wl/js/folder-listing.js]: Better to extract this out as a variable, especially as the same test is done in 2 places and they are a long way apart in the code.

Comment (buckett) [inline library/src/webapp/editor/ckextraplugins/common-wl/js/folder-listing.js]: Actually, probably better to have a `container` variable, and by default set it to be equal to `html` but if we are showing the root folder set container equal to `ul`. That way you keep the switching all in the same place.

File: library/src/webapp/editor/ckextraplugins/common-wl/js/folder-listing.js [hunk 2]
```diff
@@ -102,7 +114,12 @@ $.fn.folderListing = function(options) {
       }
 
       li.append(a);
-      html.append(li);
+      if(settings.displayRootDirectory && urlRoot === root){
```
Comment (buckett) [inline library/src/webapp/editor/ckextraplugins/common-wl/js/folder-listing.js]: Then here just do `container.append(li)` and it doesn't matter how you got here.

File: library/src/webapp/editor/ckextraplugins/common-wl/js/folder-listing.js (hunk 1 above)
Comment (ouit0408) [inline library/src/webapp/editor/ckextraplugins/common-wl/js/folder-listing.js]: when we are showing root folder we want both 'html' and 'ul', as this 'ul' is new list for the root, I am not sure if we can use just one variable

**Comment by ouit0408** [comment]
Implemented review comments and have pushed the code .

**Pull request merged**